fixed-requirements.txt

```diff
@@ -26,8 +26,8 @@ lockfile==0.12.2
 # >=0.23 was from jinja2
 MarkupSafe<2.1.0,>=0.23
 mongoengine==0.23.0
-# networkx v2.6 does not support Python3.6.  Update networkx to match orquesta
-networkx>=2.5.1,<2.6
+# required by orquesta (networkx<2.6 for py3.6, networkx<3 for py3.8)
+networkx<3
```

Comment:
@armab what happens if you change networkx here to match orquesta, e.g. >=2.6 & < 3
and then re-generate the requirements files
and then re-generate pants.

Wondering if that is why it complains that it can't rectifiy.

Reply:
Though also orquesta looks a bit more complicated...
requirements.txt:networkx>=2.5.1,<2.6; python_version < '3.7'
requirements.txt:networkx>=2.6,<3; python_version >= '3.7'

So we have different dependencies for different python versions, I wonder if the pants generate lockfiles copes with that?

Reply:
@amanda11  Yeah, I tried that in a previous step.
See the build failures here: https://github.com/StackStorm/st2/pull/6050#discussion_r1372367633

Outside of build issues I've found later that pip under EL7 didn't work as expected with the environment markers:
https://github.com/StackStorm/st2/pull/6050#issuecomment-1781768549

After all the tries in `st2-packages` and `st2` the current approach is the way of less resistance and at least I got the packages build working (with the https://github.com/StackStorm/st2/pull/6050#issuecomment-1782933536 fix)

Reply:
> I wonder if the pants generate lockfiles copes with that?

On the pants lockfile side, I saw https://github.com/StackStorm/st2/blob/8f6cb468ff50a827ce50c9be6ba78b4ac90da263/lockfiles/st2.lock#L134
so technically it may support things, but I experienced issues with pip too, so 🤷‍♂️

Reply:
pants does not support using the legacy resolver. I'm not sure why the legacy resolver is required in this case, as it seems clear to me. I'll have to play with it, hopefully this week.

Reply:
@cognifloyd Thanks for looking into it 👍  Can't wait and I'm looking forward to merge this PR so I can continue updating the dependencies for the `v3.8.1` patch.

---
Some context on that.
For some unknown reason, the new pip resolver went into an infinite loop resolving the dependencies: https://app.circleci.com/pipelines/github/StackStorm/st2/4158/workflows/11838dac-5adc-4554-9a87-f7bf506bedb6/jobs/15940?invite=true#step-109-978547_123 and couldn't process that `networkx` with conditional `python_version` env markers failing the EL7 build:

```
[package: st2] [18:59:15]      ERROR: Cannot install orquesta==1.6.0 and st2==3.9.dev0 because these package versions have conflicting dependencies.
[package: st2] [18:59:15]      
[package: st2] [18:59:15]      The conflict is caused by:
[package: st2] [18:59:15]          st2 3.9.dev0 depends on networkx
[package: st2] [18:59:15]          orquesta 1.6.0 depends on networkx<3 and >=2.6
```

In this ^^ case `orquesta 1.6.0 depends on networkx<3 and >=2.6` is a wrong condition chosen, as `EL7` runs on py3.6 and should choose a different networkx for dependency. So that might be a bug in pip.

I've tried to use [the latest pip version available for py3.6](https://github.com/StackStorm/st2-packages/pull/728/commits/28ac7f79677accc4e2209702fc44a2da1cff7364), but it didn't help.
From the pip changelog there might be a relevant fix to try: ([When checking for conflicts in the build environment, correctly skip requirements containing markers that do not match the current environment.](https://pip.pypa.io/en/stable/news/#id210)) available in pip `22.1`, but we can't use it because py3.6 support was dropped in pip `22.0`.

---

My thinking that's the worst case, for the upcoming st2 `v3.8.1` patch we'll live with it and take what worked in this PR  that works with `st2-packages`. The new pants build is not going to go live for this patch anyway.
In st2 `v3.9.0` we can deprecate python 3.6, switch to a new pip dependency resolver, latest pip and remove python env markers alltogether. That'll be quite an exercism that should potentially fix some bugs and technical debt 😈